### PR TITLE
photo-index: index photos by sources too

### PR DESCRIPTION
**Pull request by aboodman**
Ignore first three commits as they are being reviewed separately.

### CODE REVIEW

**Comment by zcstarr** [comment]
jenkins: test this

File: samples/go/photo-index/main.go [hunk 1]
```diff
@@ -171,11 +187,13 @@ func index() (win bool) {
...
-		"byFace":       byFace.Build(),
-		"tagsByCount":  stringsByCount(db, tagCounts),
-		"facesByCount": stringsByCount(db, faceCounts),
+		"byDate":         byDate.Build(),
+		"byFace":         byFace.Build(),
+		"bySource":       bySource.Build(),
+		"byTag":          byTag.Build(),
+		"facesByCount":   stringsByCount(db, faceCounts),
+		"sourcesByCount": stringsByCount(db, sourceCounts),
+		"tagsByCount":    stringsByCount(db, tagCounts),
```
Comment (kalman) [inline samples/go/photo-index/main.go]: Will you want to get the N most common tags for any reason? In which case you should add a "sourcesByCount" here as well.

Comment (aboodman) [inline samples/go/photo-index/main.go]: I'm confused. Line 195 adds `sourcesByCount`.

Comment (kalman) [inline samples/go/photo-index/main.go]: Err, I don't know what I was thinking then.

**Pull request merged**